python/pyarrow/_substrait.pyx

```diff
@@ -0,0 +1,79 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# cython: language_level = 3
+from cython.operator cimport dereference as deref
+
+from pyarrow import Buffer
+from pyarrow.lib cimport *
+from pyarrow.includes.libarrow cimport *
+from pyarrow.libarrow_substrait cimport *
+
+
+def run_query(plan):
```

Comment:
I have experimental PyArrow/Substrait code that runs a Substrait plan and returns types other than `RecordBatchReader`. Since there are other return types of interest, I'd suggest including something like "read" in this method's name, e.g. `run_query_as_reader`.

Reply:
Good point, although we haven't finalized this API yet. There are extensions to this API. Hopefully we can finalize the terminology when we get the full API integrated.